pkg/llm-d-inference-sim/tools_utils.go

```diff
@@ -0,0 +1,337 @@
+/*
+Copyright 2025 The llm-d-inference-sim Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package llmdinferencesim
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/santhosh-tekuri/jsonschema/v5"
+)
+
+func countTokensForToolCalls(toolCalls []toolCall) int {
+	numberOfTokens := 0
+	for _, tc := range toolCalls {
+		numberOfTokens += len(strings.Fields(tc.ID))
+		numberOfTokens += len(strings.Fields(tc.Type))
+		numberOfTokens += len(strings.Fields(*tc.Function.Name))
+		numberOfTokens += len(strings.Fields(tc.Function.Arguments))
+	}
+
+	return numberOfTokens
+}
+
+var fakeStringArguments = []string{
+	`testing`,
+	`hello`,
+	`Boston`,
+	`sunny`,
+	`temperature`,
+	`cloudy`,
+	`question`,
+	`Yorick`,
+	`silence`,
+	`lifetime`,
+}
+
+func getStringArgument() string {
+	index := rand.New(rand.NewSource(time.Now().UnixNano())).Intn(len(fakeStringArguments))
+	return fakeStringArguments[index]
+}
+
+func generateToolArguments(tool tool) (map[string]any, error) {
+	arguments := make(map[string]any)
+	properties, _ := tool.Function.Parameters["properties"].(map[string]any)
+
+	required := make(map[string]bool)
+	requiredParams, ok := tool.Function.Parameters["required"]
+	if ok {
+		requiredArray, _ := requiredParams.([]any)
+		for _, requiredParam := range requiredArray {
+			param, _ := requiredParam.(string)
+			required[param] = true
```

Comment:
change to `required[param] = struct{}{}`

Reply:
done